# HG changeset patch
# User jwe
# Date 750544046 0
#      Wed Oct 13 20:27:26 1993 +0000
# Node ID 8dc079be9dc5bc07e74ac6c29fc53249b6f6f866
# Parent  d46c6323a19016610abf2f0165e65fcae4f1d374
[project @ 1993-10-13 20:27:26 by jwe]
(qpsol): Don't try to create the linear constraint object before
checking array sizes.  Wouldn't exception handling be nice?

diff --git a/src/qpsol.cc b/src/qpsol.cc
--- a/src/qpsol.cc
+++ b/src/qpsol.cc
@@ -137,21 +137,21 @@ Handle all of the following:
     }
 
   if (nargin == 7 || nargin == 9)
     {
       ColumnVector lub = args[nargin-1].to_vector ();
       Matrix A = args[nargin-2].to_matrix ();
       ColumnVector llb = args[nargin-3].to_vector ();
 
-      LinConst linear_constraints (llb, A, lub);
-
       if (! linear_constraints_ok (x, llb, A, lub, "qpsol", 1))
 	return retval;
 
+      LinConst linear_constraints (llb, A, lub);
+
       if (nargin == 9)
 	{
 	  // 3. qpsol (x, H, c, lb, ub, llb, A, lub)
 
 	  QPSOL qp (x, H, c, bounds, linear_constraints);
 	  soln = qp.minimize (objf, inform, lambda);
 	}
       else
